ckanext/googleanalytics/commands.py

```diff
@@ -109,129 +113,205 @@ def load_analytics(self, args):
+                        self.log.warning("Couldn't find resource %s" % resource_url)
+                        continue
+                    for visit_date, count in visits.iteritems():
+                        ResourceStats.update_visits(resource.id, visit_date, count)
+                        self.log.info("Updated %s with %s visits" % (resource.id, count))
+                else:
+                    package_name = identifier[len(PACKAGE_URL):]
+                    if "/" in package_name:
+                        self.log.warning("%s not a valid package name" % package_name)
+                        continue
+                    item = model.Package.by_name(package_name)
+                    if not item:
+                        self.log.warning("Couldn't find package %s" % package_name)
+                        continue
+                    for visit_date, count in visits.iteritems():
+                        PackageStats.update_visits(item.id, visit_date, count)
+                        self.log.info("Updated %s with %s visits" % (item.id, count))
+
+        if querytype == 'visitorlocation':
+            print 'got into saving location'
+            for location, visits_collection in data[querytype].items():
+                visits = visits_collection.get('visits', {})
                 for visit_date, count in visits.iteritems():
-                    PackageStats.update_visits(item.id, visit_date, count)
-                    self.log.info("Updated %s with %s visits" % (item.id, count))
-        model.Session.commit()
-
-    def get_ga_data(self, start_date=None):
-        """
-        Get raw data from Google Analytics for packages and
-        resources for the start date given as parameter or last time since database was updated and 2 days more
+                    AudienceLocationDate.update_visits(location, visit_date, count)
+                    self.log.info("Updated %s on %s with %s visits" % (location, visit_date, count))
 
-        Returns a dictionary like::
 
-           {'identifier': {'visits':3, 'visit_date':<time>}}
-        """
+        model.Session.commit()
+        self.log.info("Successfully saved analytics query of type: %s" % querytype)
+
+    def query_resolvers(self, querytype, results, data):
+        '''
+        formats results and returns a dictionary like:
+        {
+            'package': { 'cool-dataset-name': { 'visits': { 2019-02-24: 500, ... } }, ... },
+            'visitorlocation': { 'Finland': { 'visits': { 2019-02-24: 500, ... } }, ... }
+        }
+        '''
+        if querytype == 'package':
+            if 'rows' in results:
+                for result in results.get('rows'):
+                    # this is still specific for packages query
+                    package = result[0]
+                    # removes /data/ from the url
+                    if not package.startswith(PACKAGE_URL):
+                        package = '/' + '/'.join(package.split('/')[2:])
```

Comment:
Maybe instead of this, check if `package.startswith('/data')`, if it does `package = package[len('/data'):]`. the above can fail with unexpected URLs.